xskillscore/core/comparative.py

```diff
@@ -127,6 +129,12 @@ def _categorical_metric(observations, forecasts, dim):
 
     if isinstance(dim, str):
         dim = [dim]
+    elif dim is None:
```

Comment:
how come this was needed? Is there a test for it?

Reply:
`mypy` complained without. the `dim==None` path wasnt described before